providers/flagsmith/src/test/java/dev.openfeature.contrib.providers.flagsmith/FlagsmithProviderTest.java

```diff
@@ -0,0 +1,105 @@
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+public class FlagsmithProviderTest {
+
+    public static MockWebServer mockFlagsmithClient;
+
+    final QueueDispatcher dispatcher = new QueueDispatcher() {
+        @SneakyThrows
+        @Override
+        public MockResponse dispatch(RecordedRequest request) {
+            if (request.getPath().startsWith("/flags/")) {
+                return new MockResponse()
+                    .setBody(readMockResponse("environment_flags_response.json"))
+                    .addHeader("Content-Type", "application/json");
+            }
+            return new MockResponse().setResponseCode(404);
+        }
+    };
+
+    private static Stream<Arguments> provideKeysForFlagResolution() {
+        return Stream.of(
+            Arguments.of("true_key", "getBooleanEvaluation", Boolean.class, "true"),
+            Arguments.of("false_key", "getBooleanEvaluation", Boolean.class, "false")
+        );
+    }
+
+    @BeforeEach
+    void setUp() throws IOException {
+        mockFlagsmithClient = new MockWebServer();
+        mockFlagsmithClient.setDispatcher(this.dispatcher);
+        mockFlagsmithClient.start();
+    }
+
+    @AfterEach
+    void tearDown() throws IOException {
+        mockFlagsmithClient.shutdown();
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("provideKeysForFlagResolution")
+    void shouldResolveFlagCorrectlyWithCorrectFlagType(
+        String key, String methodName, Class<?> expectedType, String flagsmithResult) {
+        // Given
+        FlagsmithProviderOptions options = FlagsmithProviderOptions.builder()
+                                                                   .apiKey("API_KEY")
+                                                                   .baseUri(String
+                                                                       .format("http://localhost:%s/$s",
+                                                                           mockFlagsmithClient
+                                                                               .getPort()))
+                                                                   .build();
+        FlagsmithProvider flagsmithProvider = new FlagsmithProvider(options);
+
+        Object result = null;
+        EvaluationContext evaluationContext = new MutableContext();
+
+        // When
+        Method method = flagsmithProvider.getClass()
+                                         .getMethod(methodName, String.class, expectedType, EvaluationContext.class);
+        result = method.invoke(flagsmithProvider, key, null, evaluationContext);
+
+        // Then
+        ProviderEvaluation<Object> evaluation = (ProviderEvaluation<Object>) result;
+        assertEquals(flagsmithResult, evaluation.getValue());
+        assertNull(evaluation.getErrorCode());
+    }
+
+    @Test
+    void shouldNotResolveFlagIfFlagIsInactiveInFlagsmithInsteadUsingDefaultValue() {
```

Comment:
As above, I'm not sure why this is the case?

Reply:
This is the behaviour according to the spec